lib/src/ui/common/message_provider.dart

```diff
@@ -0,0 +1,3 @@
+abstract class MessageProvider {
+  String getMessage(Object error);
```

Comment:
ten object to nie będzie to zawsze AllTheNewsException? Jak tak to mógłby być zamiast Object. Klasa mogłaby się nazywać `ErrorMessageProvider` skoro przyjmuje jako argument metody `error`

Reply:
celowo w common, bo nie jest powiedziane, że tylko errory będziemy mapować na stringi, z takiego samego powodu `AllTheNewsException` dajemy do common. Za to nazwa parametru nie powinna się nazywać `error`, tylko jakiś `identifier`

Reply:
w takim razie przyznam, że średnio mi się podoba przepychanie Object jako parametr, już lepiej jakby był generykiem:
`MessageProvider<T> { String getMessage(T object) }`